lib/github_changelog_generator/gitlab_fetcher.rb

```diff
@@ -0,0 +1,449 @@
+    end
+
+    # Fetch event for all issues and add them to 'events'
+    #
+    # @param [Array] issues
+    # @return [Void]
+    def fetch_events_async(issues)
+      i       = 0
+      threads = []
+      options = { }
+      if issues.empty?
+        return
+      end
+      options[:target_type] = issues.first["merged_at"].nil? ? "issue" : "merge_request"
+      issue_events = []
+      @client.project_events(@project_id, options).auto_paginate do |event|
+        event = stringify_keys_deep(event.to_hash)
+        # gitlab to github
+        event["event"] = event["action_name"]
+        issue_events.push(event)
+      end
+      # p issue_events
+
+      issues.each_slice(MAX_THREAD_NUMBER) do |issues_slice|
+        issues_slice.each do |issue|
+          threads << Thread.new do
+            issue["events"] = []
+            issue_events.each do |new_event|
+              if issue["id"] == new_event["target_id"]
+                if new_event["action_name"].eql? "closed"
+                  issue["closed_at"] = issue["closed_at"].nil? ? new_event["created_at"] : issue["closed_at"]
+                end
+                issue["events"].push(new_event)
+              end
+            end
+            print_in_same_line("Fetching events for #{options[:target_type]}s: #{i + 1}/#{issues.count}")
+            i += 1
+          end
+        end
+        threads.each(&:join)
+        threads = []
+      end
+
+      # to clear line from prev print
+      print_empty_line
+
+      Helper.log.info "Fetching events for issues and PR: #{i}"
+    end
+
+    # Fetch comments for PRs and add them to "comments"
+    #
+    # @param [Array] prs The array of PRs.
+    # @return [Void] No return; PRs are updated in-place.
+    def fetch_comments_async(prs)
+      threads = []
+
+      prs.each_slice(MAX_THREAD_NUMBER) do |prs_slice|
+        prs_slice.each do |pr|
+          threads << Thread.new do
+            pr["comments"] = []
+            @client.merge_request_notes(@project_id, pr["number"]) do |new_comment|
+              new_comment = stringify_keys_deep(new_comment.to_hash)
+              new_comment["body"] = new_comment["description"]
+              pr["comments"].push(new_comment)
+            end
+            pr["comments"] = pr["comments"].map { |comment| stringify_keys_deep(comment.to_hash) }
+          end
+        end
+        threads.each(&:join)
+        threads = []
+      end
+      nil
+    end
+
+    # Fetch tag time from repo
+    #
+    # @param [Hash] tag GitHub data item about a Tag
+    #
+    # @return [Time] time of specified tag
+    def fetch_date_of_tag(tag)
+      Time.parse(tag["commit"]["committed_date"])
+    end
+
+    # Fetch and cache comparison between two github refs
+    #
+    # @param [String] older The older sha/tag/branch.
+    # @param [String] newer The newer sha/tag/branch.
+    # @return [Hash] Github api response for comparison.
```

Comment:
Mentions Github api - it's probably GitLab client-related content which is produced here.